File: Practice(Beginner)/C++/Puppy and Sum.cpp
// Source: https://www.codechef.com/problems/PPSUM

#include <bits/stdc++.h>
using namespace std;

long long int sumTillN(long long int n){
    return (n*(n+1))/2;
}

int main() {
    long long int t, iter , n;
    cin >> t;
    while (t--){
        cin >> iter >> n;
        while (iter--){
            n = sumTillN(n);
        }
        cout << n << endl;
    }
    return 0;
}

File: Practice(Beginner)/Python3/Puppy_and_Sum.py
# Source: https://www.codechef.com/problems/PPSUM

from functools import reduce
it = int(input())
for j in range(it):
    a,b=map(int,input().split())
    for i in range(a):
        b = reduce(lambda x,y: x+y,list(range(1,b+1)))
    print(b)
